app/controllers/text_components_controller.rb

```diff
@@ -86,11 +90,19 @@ def set_index_data
       @text_components = TextComponent.includes(:triggers, :question_answers, :channels)
 
       filter_text_components
+      
+      @trigger_groups = @text_components.order('from_day').group_by { |t| t.trigger_ids }
+      @trigger_groups = @trigger_groups.map { |trigger_ids, components| [ Trigger.find(trigger_ids), components ] }.to_h
+      @trigger_groups = @trigger_groups.sort do |a,b|
+        if(a.first.first && b.first.first)
+          a.first.first.name.downcase <=> b.first.first.name.downcase
```

Comment:
Hahah okay, let me try to explain…

The table of text components (or trigger groups) should be sorted by the name of the trigger. `@trigger_groups` is a has with an array of trigger_ids as the key and a two-element array containing a collection of triggers and the matching text components.

So when sorting `a.first.first` is refering to the first trigger in the list of triggers for that trigger group. Phew, not very elegant – I’ll clean this up a little tomorrow.